### PR TITLE
WebSocket actions improvement

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
I was looking to get posts as soon as the WS reconnects and fixing up a bit the reconnect logic, and I ended up modifying a bit of code to make the WebSocket actions a bit better.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-12832
https://mattermost.atlassian.net/browse/MM-13039

### CODE REVIEW

**Comment by enahum** [comment]
Ticket to move the dispatcher here https://mattermost.atlassian.net/browse/MM-13039

File: src/client/websocket_client.js [hunk 1]
```diff
@@ -172,35 +174,35 @@ class WebSocketClient {
...
+                } else if (this.eventAction) {
+                    dispatch(this.eventAction(msg));
                 }
             };
         });
     }
 
-    setConnectingCallback(callback) {
-        this.connectingCallback = callback;
+    setConnectingAction(action) {
```
Comment (hmhealey) [inline src/client/websocket_client.js]: This is going to break anyone else using the websocket client. I'm pretty sure we want backwards compatibility here

Comment (enahum) [reply src/client/websocket_client.js]: this is going to break anyway as we changed the way the previous callbacks now actions are called

Comment (enahum) [reply src/client/websocket_client.js]: But it should not be a problem as in reality people should be using the WS actions and not the client directly

**Comment by enahum** [comment]
@jwilander @hmhealey refactor this a bit more to remove the `dispatch` from the client

**Comment by enahum** [comment]
rebased with the latest master that includes the dialogs

**Comment by enahum** [comment]
I'm not QA but I did test it using the mobile app and it looked good, but yeah, I can submit a PR to the mobile repo and build it so QA can test

**Pull request merged**